site/content/extend/plugins/migration.md

```diff
@@ -0,0 +1,213 @@
+
+The plugin entry point used to be registered by directly manipulating a global variable:
+
+```js
+window.plugins['my-plugin'] = new MyPlugin();
+```
+
+Instead, use the globally exported `registerPlugin` method:
+
+```js
+window.registerPlugin('my-plugin', new MyPlugin());
+```
+
+### Externalizing Dependencies
+
+The plugins beta suggested relying on the global export of common libraries from the web app:
+
+```js
+const React = window.react;
+```
+
+While this remains supported, it is more natural to leverage Webpack [Externals](https://webpack.js.org/configuration/externals/). Configure this in your `.webpack.config.js`:
+
+```
+module.exports = {
+    // ...
+    externals: {
+        react: 'react',
+    },
+    // ...
+};
+```
+
+and then import your modules naturally:
+
+```js
+import React from 'react';
+```
+
+### Initialization
+
+The `initialize` callback used to receive a `registerComponents` callback to configure components, post types and main menu overrides:
+
+```js
+import ChannelHeaderButton from './components/channel_header_button';
+import MobileChannelHeaderButton from './components/mobile_channel_header_button';
+import PostTypeZoom from './components/post_type_zoom';
+import {configureZoom} from './actions/zoom';
+
+class MyPlugin {
+    initialize(registerComponents) {
+         registerComponents(
+             {ChannelHeaderButton, MobileChannelHeaderButton}, 
+             {custom_zoom: PostTypeZoom}, 
+             {
+                 id: 'zoom-configuration',
+                 text: 'Zoom Configuration',
+                 action: configureZoom,
+             },
+        );
+    }
+}
+```
+
+The `initialize` callback now receives an instance of the plugin [registry](/extend/plugins/webapp/reference/#registry). In some cases, the registry's API now requires a more discrete breakdown of the registered component to allow the web app to handle various rendering scenarios:
+
+```js
+import ChannelHeaderButtonIcon from './components/channel_header_button/icon';
+import MobileChannelHeaderButton from './components/mobile_channel_header_button';
+import PostTypeZoom from './components/post_type_zoom';
+import {startZoomMeeting, configureZoom} from './actions/zoom';
+
+class MyPlugin {
+    initialize(registry) {
+        registry.registerChannelHeaderButtonAction(
+            ChannelHeaderButtonIcon,
+            startZoomMeeting,
+            'Start Zoom Meeting',
+        );
+
+        registry.registerPostTypeComponent('custom_zoom', PostTypeZoom);
+
+        registry.registerMainMenuAction(
+            'Zoom Configuration',
+            configureZoom,
+            MobileChannelHeaderButton,
+        );
+    }
+}
+```
+
+Restructuring your plugin to use the new registry API will likely prove to be the hardest part of migrating.
+
+### Uninitialization
+
+The `uninitialize` callback used to be called `deinitialize`:
+```js
+class MyPlugin {
+    // ...
+    deinitialize() {
+        // ...
+    }
+    // ...
+}
+```
```

Comment:
We never actually released a Mattermost version that had deinitialize in it, so we can probably remove this part